### DISCUSSION TITLE
Copilot ESC key doesn't escape INSERT mode in IdeaVim. Have to press ESC twice every time.

### DISCUSSION THREAD

**Discussion by gak** [community/community#72915] [Copilot Conversations] [answered]
IDE: RustRover

IdeaVim + Copilot should need only one ESC press to get out of Insert mode.

It recently changed to make you press it twice.

Reproduction:
* Vim Insert mode
* Type something to trigger Copilot suggestion.
* Press ESC.
* Copilot suggestion is cleared, but still in Insert mode.
* Have to press ESC again to exit Insert mode.

As a Vim user this is frustrating, as ESC should immediately exit out of Insert mode.

Potentially same issues:
https://youtrack.jetbrains.com/issue/VIM-2772/Regression-of-Escape-key-in-insert-mode-closes-autocomplete-dropdown-without-exiting-insert-mode
https://youtrack.jetbrains.com/issue/VIM-3156/ESC-do-not-work-again

**Comment by gak** [comment]
The update to 2.6.3 seems to have reverted the problem as the changelog says:

> 2.6.0 and 2.6.1 releases are broken. Version 2.6.3 reverts IdeaVim plugin to the working state as for 2.5.1.

**Reply by gak** [reply]
It just started happening again in v2.7.0!

**Reply by gak** [reply]
Reverting to 2.6.3 works again. Copilot 1.4.0.3775-nightly. RustRover 2023.3 EAP.

**Comment by am1r404** [comment]
still doesn't work after update

**Reply by gak** [reply]
@Amirkhan98 I've unmarked the answer for you so more 👀 can see it.

It has been working for me all day. I'm not sure what else it could be.

**Comment by Teddy12090** [comment, marked as answer]
I have the same issue. I found that it is caused by hotkey settings. In my case, the `ese` is linked to **Escape** and  **Copilot: Hide Completions in Editor**. After I removed the relation on the Copilot action, it works correctly.

<img width="993" alt="image" src="https://github.com/community/community/assets/3054877/0d0b04d3-7fac-40ec-9a3e-5c249564c324">

**Reply by gak** [reply]
Thanks, that seems to solve it for me.

**Reply by jongoestoeleven** [reply]
This issue resurfaced and the KeyMap entry for Copilot: Hide Completions in Editor no longer exists.

**Comment by PRSFM** [comment]
Having the same issue nowadays and the hide completions no longer exists to unbind. Really annoying
